addons/docs/src/mdx/mdx-compiler-plugin.js

```diff
@@ -249,7 +249,14 @@ function getExports(node, counter, options) {
     if (CANVAS_REGEX.exec(value)) {
       // Canvas/Preview, possibly containing multiple stories
       const ast = parser.parseExpression(value, { plugins: ['jsx'] });
-      return { stories: genCanvasExports(ast, counter) };
+
+      const canvasExports = genCanvasExports(ast, counter);
+
+      const { code } = generate(ast, {});
+      // eslint-disable-next-line no-param-reassign
+      node.value = code;
```

Comment:
What's going on here? Feels like it could do with a comment at the very least.

Reply:
Will fix in the next PR